dali/kernels/signal/moving_mean_square_gpu_test.cc

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/signal/moving_mean_square_gpu.h"
+#include <gtest/gtest.h>
+#include <vector>
+#include "dali/core/cuda_event.h"
+#include "dali/kernels/dynamic_scratchpad.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+namespace test {
+
+
+template<class InputType>
+class MovingMeanSquareGPU : public ::testing::Test {
+ public:
+  using In = int;
+  using Out = float;
+
+  TestTensorList<In> in_;
+  TestTensorList<Out> out_;
+
+  void SetUp() final {
+    int nsamples = 4;
+
+    TensorListShape<> sh = {{30, }, {1000, }, {50, }, {40, }};
+    in_.reshape(sh);
+    out_.reshape(sh);
+
+    std::mt19937 rng;
+    UniformRandomFill(in_.cpu(), rng, 0.0, 1.0);
+  }
+
+  /**
+   * @brief Naive calculation. Square and sum over a region
+   */
+  Out mean_squared(In *start, In *pos, int window_size) {
```

Comment:
```suggestion
  Out mean_square(In *start, In *pos, int window_size) {
```
is the typical term.
`mean_squared` sounds too like a "squared mean".